### ISSUE TITLE
CMUDict CNTK file reading problem

### ISSUE THREAD

**Issue by RafaelDrumond** [Microsoft/CNTK#832]
(I couldn't re-open  issue #768 so I had to create a new one)
Hey guys, I have renamed the fileExt variable as @raaaar said. Thanks, it looks like I am getting past that error. But I am getting new errors. Can you guys lend me a hand again?

It seems to be a problema that's ocurring while reading the ctf file. Could it be my CNTK version (1.5)?

The output was a bit big, so I can't paste as comment.
[G2P_train.txt](https://github.com/Microsoft/CNTK/files/466356/G2P_train.txt)

Thanks in advance!

**Comment by RafaelDrumond** [comment]
Ok, I think I found out the two main problems:

At line 92 we have this defining the readers inputs:
`cntkReaderInputDef             = [ rawInput = [ alias = "s" ; dim = $inputVocabSize$ ; format = "sparse" ] ; rawLabels = [ alias = "t" ;  dim = $labelVocabSize$ ;  format = "sparse" ] ]`

however, in the ctf files we have input as "S0" and labels as "S1", changing line 92 should help if we change it to

`cntkReaderInputDef             = [ rawInput = [ alias = "S0" ; dim = $inputVocabSize$ ; format = "sparse" ] ; rawLabels = [ alias = "S1" ;  dim = $labelVocabSize$ ;  format = "sparse" ] ]`

However, another error shows up, regarding the dimension of the label sparse vector in the ctf file exceeding the size defined in the config file which is:

`inputVocabSize = 29     # 26 letters plus start, end, apostrophe
labelVocabSize = 41     # 39 phonemes (~AX missing), plus start and end symbol (in index 0)`

I believe that the numbers are correct however the phonemes and graphemes are being mapped in the same vector which has size 69 and the labels of the train/validation files are assigned to this mapping vector (should I call it this way?).

In order to work, I had to change the labels dimension to a bigger value in order to hold the whole vector, but that would leave some labels in "blank" right?

Any thoughts on this?
